### PR TITLE
Add utility functions to ReferenceCell namespace

### PR DESCRIPTION
**Pull request by peterrum**
These are:
- `create_coarse_triangulation()`
- `get_default_linear_mapping()`
- `get_default_quadrature()`

Needed by #11422.

### CODE REVIEW

File: source/grid/reference_cell.cc [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        Assert(false, ExcNotImplemented());
+      }
+  }
+
+
+  template <int dim, int spacedim>
+  const Mapping<dim, spacedim> &
+  get_default_linear_mapping(const Type &reference_cell)
+  {
+    if (reference_cell == get_hypercube(dim))
```
Comment (bangerth) [inline source/grid/reference_cell.cc:114]: same here with the assertion

File: include/deal.II/grid/reference_cell.h [hunk 2]
```diff
@@ -183,6 +192,32 @@ namespace ReferenceCell
...
+  get_default_linear_mapping(const Type &reference_cell);
+
+  /**
+   * Return a default quadrature matching the given reference cell (MappingQ1
+   * for hypercube cells and MappingFE else).
+   */
+  template <int dim>
+  Quadrature<dim>
+  get_default_quadrature(const Type &   reference_cell,
+                         const unsigned n_points_1D);
```
Comment (bangerth) [inline include/deal.II/grid/reference_cell.h:219]: Can you document the second argument?

Is there a "default quadrature" for cell types? It's not clear to me why that would be a Gauss quadrature formula. We often use those, but it's not the only correct choice. Maybe `get_gauss_type_quadrature()` would be a better name? Or `get_gauss_quadrature()`?

**Review by bangerth — CHANGES_REQUESTED** [review]
Looks good in principle.

I wished that in hindsight we had made the reference cell type an actual class, not just an enum. You're introducing a lot of functions that take as argument a `RerenceCell::Type` object. If we had made that an actual class, all of these functions could have been member functions just like we do for `GeometryInfo`. But I'm afraid that ship has sailed :-(

File: include/deal.II/grid/reference_cell.h [hunk 3]
```diff
@@ -183,6 +192,32 @@ namespace ReferenceCell
...
+   * reference cell.
+   */
+  template <int dim, int spacedim>
+  void
+  create_coarse_triangulation(Triangulation<dim, spacedim> &tria,
+                              const Type &                  reference_cell);
+
+  /**
+   * Return a default linear mapping matching the given reference cell (QGauss,
+   * Simplex::QGauss, Simplex::QGaussPyramid, Simplex::QGaussWedge).
```
[the remark this answers is not in the document]
Comment (peterrum) [reply include/deal.II/grid/reference_cell.h:206]: Oh yes. Good catch!

File: source/grid/reference_cell.cc [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+DEAL_II_NAMESPACE_OPEN
+
+namespace ReferenceCell
+{
+  template <int dim, int spacedim>
+  void
+  create_coarse_triangulation(Triangulation<dim, spacedim> &tria,
+                              const Type &                  reference_cell)
+  {
+    if (reference_cell == get_hypercube(dim))
```
[the remark this answers is not in the document]
Comment (peterrum) [reply source/grid/reference_cell.cc:37]: The same answer: https://github.com/dealii/dealii/pull/11423#discussion_r549687960

File: include/deal.II/grid/reference_cell.h (hunk 2 above)
Comment (peterrum) [reply include/deal.II/grid/reference_cell.h:219]: I went with `get_gauss_type_quadrature()`.

File: source/grid/reference_cell.cc [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    else if (reference_cell == Type::Tet)
+      {
+        std::vector<Point<spacedim>> vertices = {
+          {Point<spacedim>(0.0, 0.0, 0.0),
+           Point<spacedim>(1.0, 0.0, 0.0),
+           Point<spacedim>(0.0, 1.0, 0.0),
+           Point<spacedim>(0.0, 0.0, 1.0)}};
+
+        std::vector<CellData<dim>> cells(1);
+        cells[0].vertices = {0, 1, 2, 3};
```
[the remark this answers is not in the document]
Comment (peterrum) [reply source/grid/reference_cell.cc:67]: This is not possible. Here a field is initialized.

File: source/grid/reference_cell.cc [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+  {
+    if (reference_cell == get_hypercube(dim))
+      {
+        GridGenerator::hyper_cube(tria, 0, 1);
+      }
+    else if (reference_cell == Type::Tri)
+      {
+        std::vector<Point<spacedim>> vertices(3);
+
+        vertices[0][0] = 0.0;
```
[the remark this answers is not in the document]
Comment (peterrum) [reply source/grid/reference_cell.cc:45]: > Is it necessary to duplicate this information, or could the vertex locations already be obtained from some other members of the existing classes/namespaces?

There is no place yet where to query this information. For linear elements, these are exactly the same points as stored in `Simplex::Poly`, `Simplex::FE_Pyramid`, and `Simplex::FE_Wedge` as support points (but this is somewhat a coincidence). 

We can move the information in a follow-up PR into `ReferenceCell::internal::Info`.

**Comment by peterrum** [comment]
@bangerth Thanks for the review. I have addressed your comments!

**Comment by peterrum** [comment]
@bangerth I have rebased this branch and have add the asserts `AssertDimension(dim, get_dimension(reference_cell));`. Let's proceed here so that I can update #11428 and #11422 so that we can run p- and h-multigrid on simplex meshes.

**Review by bangerth — APPROVED** [review]

**Pull request merged**